cpp/cmake_modules/BuildUtils.cmake

```diff
@@ -97,7 +97,30 @@ function(arrow_create_merged_static_lib output_target)
   endforeach()
 
   if(APPLE)
-    set(BUNDLE_COMMAND "libtool" "-no_warning_for_no_symbols" "-static" "-o"
+    # The apple-distributed libtool is what we want for bundling, but there is
+    # a GNU libtool that has a namecollision (and happens to be bundled with R, too).
+    # We are not compatible with GNU libtool, so we need to avoid it.
+    # TODO: use a VALIDATOR when we require cmake >= 3.25
+
+    # check in the obvious places first to find Apple's libtool
+    find_program(LIBTOOL_MACOS libtool
+                 PATHS /usr/bin /Library/Developer/CommandLineTools/usr/bin
+                 NO_DEFAULT_PATH)
+
+    # if it's not found in obvious places, check on the standard path
+    if(LIBTOOL_MACOS-NOTFOUND)
+      set(LIBTOOL_MACOS "libtool")
+    endif()
```

Comment:
I believe this could technically be `find_program(LIBTOOL_MACOS libtool)` and that would find libtool along the search path, but be a no-op if line 106 above it found libtool.

Does that look better? Or just an obfuscated version of this?